### PR TITLE
#21454 Copying server.xml to the right path

### LINKED ISSUE #21454: Clean up docker image
We need to clean up how our docker image runs.  This includes

- [x] removing `dockerize` and the dockerized templates 
- [x] removing the init/waiting scripts for db and elasticsearch
- [x] using only dotCMS environmental variables to configure docker.  This means that for the most part, configuring the DB, ES, and dotCMS options should be the same whether you are running locally or in docker.
- [x] Removing hazelcast customizations - if hazelcast is to be used, admins should mount a custom hazelcast.xml file
- [x] Centralizing all config variables into a `config-defaults.sh`
- [x] Allow for APR to init and run
- [x] Incorporate the changes from the other docker issues, e.g.: 
     - #21440
     - #21063
     - #20892

### LINKED ISSUE #21572: allow Dockerfile to build with gid=0
When building a docker image, we give the user the ability to specify the USER_NAME, UID and GID of the user to run as.

```
ARG USER_UID="65001"
ARG USER_GID="65001"
ARG USER_NAME="dotcms"

# Creating default non-user 
# the useradd 
RUN groupadd -g $USER_GID $USER_NAME && \
	useradd -l -d /srv -g $USER_GID -u $USER_UID $USER_NAME

# Copy our build
COPY --from=build-cms --chown=dotcms:dotcms /srv/ /srv/
COPY --chown=dotcms:dotcms ROOT/ /
```

The issue is that in some cases, for example in openshift, you need to be able to specify a `USER_ID=65001` with the ROOT group for GID, e.g. `USER_GID=0`.    This breaks our Dockerfile build because we explicitly add whatever group has been specified.

Also, if you specify a `USER_NAME`, you can see that we do not use it when we are `chown`ing the files, which will break as well.

### PR DESCRIPTION
**Pull request by nollymar**
With the cleanup, some required steps were removed

### CODE REVIEW

**Review by jdotcms — APPROVED** [review]

**Review by fabrizzio-dotCMS — APPROVED** [review]

File: docker/dotcms/ROOT/srv/20-copy-overriden-files.sh [hunk 1]
```diff
@@ -2,5 +2,9 @@
 
 set -e
 
+[[ -f /srv/TOMCAT_VERSION ]] && TOMCAT_VERSION=$( cat /srv/TOMCAT_VERSION )
+
 cp /srv/OVERRIDE/tomcat/conf/*      $TOMCAT_HOME/conf/
+mv $TOMCAT_HOME/conf/server-${TOMCAT_VERSION}.xml $TOMCAT_HOME/conf/server.xml && \
```
Comment (wezell) [inline docker/dotcms/ROOT/srv/20-copy-overriden-files.sh:8]: I think the fix is to change the filename from `server-${TOMCAT_VERSION}.xml` to `server.xml`.  With the Dockerfile in the core repo, we do not need to support multiple versions of tomcat in any given branch and can remove all that  conditional logic.

**Review by wezell — CHANGES_REQUESTED** [review]
see my comment.  We need to remove any conditional tomcat version logic as in any given branch we know the exact path/version of tomcat

**Comment by nollymar** [comment]
> see my comment. We need to remove any conditional tomcat version logic as in any given branch we know the exact path/version of tomcat

@wezell Done! Please take a look at my latest changes

File: docker/dotcms/Dockerfile [hunk 2]
```diff
@@ -51,16 +51,18 @@ RUN apt purge --allow-remove-essential -y  && \
...
 ARG USER_GID="65001"
+ARG USER_GROUP="dotcms"
 ARG USER_NAME="dotcms"
 
-# Creating default non-user 
-# the useradd 
-RUN groupadd -g $USER_GID $USER_NAME && \
-	useradd -l -d /srv -g $USER_GID -u $USER_UID $USER_NAME
+
+RUN groupadd -f -g $USER_GID $USER_NAME
```
Comment (wezell) [inline docker/dotcms/Dockerfile:58]: this needs to be changed to USER_GROUP

File: docker/dotcms/README.md [hunk 3]
```diff
@@ -22,6 +22,8 @@ Please send any thoughts or suggestions to improve our docker image to support@d
 | JAVA_VERSION  | SDKMAN java version to use | 
 | USER_UID      | User ID to run as | 
 | USER_GID      | Group ID for user | 
+| USER_NAME     | User name to run as |
+| USER_GROUP    | Group name for user |
```
Comment (wezell) [inline docker/dotcms/README.md:26]: Add what the defaults are

File: docker/dotcms/Dockerfile (hunk 2 above)
Comment (nollymar) [reply docker/dotcms/Dockerfile]: Done

File: docker/dotcms/README.md (hunk 3 above)
Comment (nollymar) [reply docker/dotcms/README.md]: Done

**Pull request merged**